src/seedsigner/gui/screens/seed_screens.py

```diff
@@ -850,11 +850,11 @@ def _run(self):
                 self.hw_button3.is_selected = True
                 self.hw_button3.render()
                 self.renderer.show_image()
-                return self.passphrase
+                return self.passphrase, None
 
             elif input == HardwareButtonsConstants.KEY_PRESS and self.top_nav.is_selected:
                 # Back button clicked
-                return self.top_nav.selected_button
+                return self.passphrase, self.top_nav.selected_button
```

Comment:
I'm pretty uncomfortable with this change to return a tuple and especially the downstream changes it creates in the related flow-based tests.

We have type hints for what we expect `View.run_screen` to return:
https://github.com/SeedSigner/seedsigner/blob/dev/src/seedsigner/views/view.py#L104

Since this is python, obviously we're not restricted to just those two types.

It's not terrible to break the existing convention in this way, but I still strongly dislike it. Though admittedly it's probably more on aesthetic / esoteric grounds; I doubt this would cause problems for us in the future. That being said, the flow-based tests are only possible because we held extreme discipline throughout the code in plenty of other areas.

(and we have a matching type hint for `FlowStep` that's no longer completely accurate: https://github.com/SeedSigner/seedsigner/blob/dev/tests/base.py#L109)

But I can't think of an alternate solution, either. Well, not a good one, at least.

Very minor improvement: return a `dict` instead of a tuple?
```python
# in the Screen:
# Completed passphrase entry...
return dict(passphrase=self.passphrase)

# BACK button selected...
return dict(passphrase=self.passphrase, is_back_button=True)
```

And then the View would ask:
```python
if "is_back_button" in ret_dict:
    ...
```

There's barely a difference using a dict vs tuples, but when in doubt I prefer the self-documenting nature of having dict keys.

---

Ultimately `Screen` probably should have returned something like a `Response` object that could handle a more complex use case like this. But that would be a huge refactor. But returning a `dict` here feels like a half-step in that direction and so is another very small nod in favor of dict over tuple.

---

AT THE VERY LEAST this Screen should include a `TODO` comment noting its unusual return value approach.

Reply:
Thanks, @kdmukai. I just pushed https://github.com/SeedSigner/seedsigner/pull/563/commits/d8b7672cf48b93eb22abdf5796b420a1000e1af4 following your suggestion for a dict approach and a TODO comment noting the unusual return value.

Maybe it's okay with that, or maybe we should include another comment regarding the type hints in the files you mentioned.